### ISSUE TITLE
ao2mo.general() gives wrong result with 4-fold symmetrized ERI

### ISSUE THREAD

**Issue by zhcui** [pyscf/pyscf#328]
Hi all,

I try to use `ao2mo.general()` to transform an (aa|bb) type integral (usually used in an unrestricted post-HF routine) to MO basis. The integral has 4-fold symmetry. However, the function seems assuming further symmetry of (aa|bb) = (bb|aa), which will give wrong result as compared with np.einsum. Here is a concrete example:

```
#!/usr/bin/env python
"""
An example to show the inconsistency of ao2mo.general() 
for (aa|bb) type of integral, if ERI symmetry is used.
"""

import numpy as np
import scipy.linalg as la
from pyscf import ao2mo

np.random.seed(1)
norb = 4 

# A 4-index eri with 4-fold symmetry
h2_s1 = np.random.random((norb, norb, norb, norb))
h2_s1 = h2_s1 + h2_s1.transpose(1,0,2,3)
h2_s1 = h2_s1 + h2_s1.transpose(0,1,3,2)
#h2_s1 = h2_s1 + h2_s1.transpose(2,3,0,1) # 8-fold symmetry does not exist 

# pack the eri to 2-index
h2_s4 = ao2mo.restore(4, h2_s1, norb)
# check the restored eri is the same as before
assert(np.allclose(ao2mo.restore(1, h2_s4, norb), h2_s1))

# 2 sets of different MO
mo_a, _ = la.qr(np.random.random((norb, norb)))
mo_b, _ = la.qr(np.random.random((norb, norb)))

# 3 possible ways to generate MO based ERI 
# from ERI with no symmetry 
eri_mo_from_s1 = ao2mo.general(h2_s1, (mo_a, mo_a, mo_b, mo_b), compact=False).reshape((norb,)*4)
# from ERI with 4-fold symmetry (2-index)
eri_mo_from_s4 = ao2mo.restore(1, ao2mo.general(h2_s4, (mo_a, mo_a, mo_b, mo_b)), norb)
# from definition, reference
eri_ref = np.einsum('ijkl, iI, jJ, kK, lL -> IJKL', h2_s1, mo_a, mo_a, mo_b, mo_b)

print "Difference between ERI from s1 and s4"
print la.norm(eri_mo_from_s4 - eri_mo_from_s1)
print "Difference between ERI from s1 and reference"
print la.norm(eri_mo_from_s1 - eri_ref)

```

Note that the 3 ways to transform a 4-fold symmetrized ERI to MO basis should give exactly the same answer, but they do not. Is it a bug?

Best,
Zhihao

**Comment by zhcui** [comment]
Hi Qiming @sunqm ,

I saw you made some commits to your local repo. Is the issue solved by the change or need more careful treatment?

Thanks!

Best,
Zhihao

**Comment by sunqm** [comment]
Hi Zhihao, it's solved in commit e8411f3

**Comment by zhcui** [comment]
Hi Qiming,

Thanks, it works now.

Best,
Zhihao
